pom.xml

```diff
@@ -195,6 +200,14 @@
         <plugin>
           <artifactId>maven-surefire-plugin</artifactId>
           <version>2.18.1</version>
+          <dependencies>
+            <dependency>
+              <!-- Force junit provider because surefire will choose the testng one if testng is detected -->
+              <groupId>org.apache.maven.surefire</groupId>
+              <artifactId>surefire-junit47</artifactId>
```

Comment:
Now I know what you mean. Basically you want the Surefire not to send JUnit class if junit=false. It is logical but there might be an unlikely extreme when the user has both testng and junit in one class. Not usual but technically possible.